tensorflow/core/kernels/training_ops_gpu.cu.cc

```diff
@@ -176,29 +175,31 @@ __device__ std::complex<T> impl_rsqrt(std::complex<T> x) {
   // due to subtraction of two close values. We have to get fancy
   root[0] = sqrt(r * ((std::is_same<T, float>::value && re * r < -0.98)
                           ? rsqrt_helper(im * im * r * r)
-                          : 1 + re * r)) *
+                          : max(T(0.0), 1 + re * r))) *
```

Comment:
Both impl_sqrt(complex<T>) and impl_rsqrt(complex<T>) look pretty bad to me. I will fix them in a separate change.

Reply:
Would you mind to take a look at https://github.com/tensorflow/tensorflow/pull/41303? Thanks!

Reply:
will drop this change from this PR

Reply:
done

Reply:
Thanks.